File: app/src/main/java/ru/nehodov/weatherforecast/utils/WeatherUtil.java
package ru.nehodov.weatherforecast.utils;

import java.text.SimpleDateFormat;
import java.time.Instant;
import java.time.LocalDate;
import java.time.ZoneId;
import java.time.format.TextStyle;
import java.util.Date;
import java.util.Locale;

public class WeatherUtil {

    /**
     * Method rounds temperature up to an integer and if temperature is positive adds plus sign
     *
     * @param tempBefore Text temperature before formatting
     * @return formatted temperature
     */
    public static String formatTemp(String tempBefore) {
        if (null == tempBefore || tempBefore.length() == 0
                || !tempBefore.matches("^-?\\d+\\.?(\\d+)?$")) { //filter for wrong formats
            return "";
        }
        try {
            double temp = Double.parseDouble(tempBefore);
            if (temp > -0.5 && temp < 0.5) {
                return "0";
            } else {
                return String.format(Locale.US, "%+.0f", temp);
            }
        } catch (NumberFormatException e) {
            return "";
        }
    }

    public static String formatDate(String unixTime) {
        if (unixTime == null || unixTime.length() == 0 || !unixTime.matches("^\\d*$")) {
            return "";
        }
        Instant instant = Instant.ofEpochSecond(Long.parseLong(unixTime));
        LocalDate date = instant.atZone(ZoneId.systemDefault()).toLocalDate();
        LocalDate now = LocalDate.now();
        if (date.equals(now)) {
            return "Today";
        } else if (date.equals(now.plusDays(1))) {
            return "Tomorrow";
        } else {
            return String.format(Locale.US, "%02d %02d %s",
                    date.getDayOfMonth(),
                    date.getMonthValue(),
                    date.getDayOfWeek().getDisplayName(TextStyle.SHORT, Locale.US));
        }
    }

    public static String formatTime(String unixTime) {
        if (null == unixTime || unixTime.length() == 0 || !unixTime.matches("^\\d*$")) {
            return "";
        } else {
            return new SimpleDateFormat("HH:mm", Locale.getDefault())
                    .format(new Date(Long.parseLong(unixTime) * 1000));
        }
    }
}

File: app/src/main/java/ru/nehodov/weatherforecast/utils/WeatherUtil.kt
package ru.nehodov.weatherforecast.utils

import java.text.SimpleDateFormat
import java.time.Instant
import java.time.LocalDate
import java.time.ZoneId
import java.time.format.TextStyle
import java.util.*

class WeatherUtil {

    companion object {
        private val CORRECT_TEMP_FORMATS = Regex("^-?\\d+\\.?(\\d+)?$")
        private val CORRECT_UNIX_TIME_FORMAT = Regex("^\\d*\$")

        /**
         * Method rounds temperature up to an integer and if temperature is positive adds plus sign
         *
         * @param tempBefore Text temperature before formatting
         * @return formatted temperature
         */
        fun formatTemp(tempBefore: String?): String {
            var tempAfter = ""
            if (!tempBefore.isNullOrEmpty() && tempBefore.matches(CORRECT_TEMP_FORMATS)) {
                    val temp = tempBefore.toDouble()
                tempAfter = if (temp in -0.5..0.5) "0" else String.format(Locale.US, "%+.0f", temp)
            }
            return tempAfter
        }

        fun formatDate(unixTime: String?): String {
            if (!unixTime.isNullOrEmpty() && unixTime.matches(CORRECT_UNIX_TIME_FORMAT)) {
                val instant: Instant = Instant.ofEpochSecond(unixTime.toLong())
                val date: LocalDate = instant.atZone(ZoneId.systemDefault()).toLocalDate()
                val now: LocalDate = LocalDate.now()
                return when (date) {
                    now -> "Today"
                    now.plusDays(1) -> "Tomorrow"
                    else -> {
                        String.format(Locale.US, "%02d %02d %s",
                                date.dayOfMonth,
                                date.monthValue,
                                date.dayOfWeek.getDisplayName(TextStyle.SHORT, Locale.US))
                    }
                }
            } else return ""
        }

        fun formatTime(unixTime: String?): String? {
            return if (unixTime.isNullOrEmpty() || !unixTime.matches(CORRECT_UNIX_TIME_FORMAT)) {
                ""
            } else {
                SimpleDateFormat("HH:mm", Locale.getDefault())
                        .format(Date(unixTime.toLong() * 1000))
            }
        }
    }
}